### ISSUE TITLE
Very bad stuttering with i5-11500 and RTX 3090

### ISSUE THREAD

**Issue by Neodoris** [Albeoris/Memoria#998]
Hi,

Today, I installed FFIX, ran it once, then installed the latest version of memoria. Inside, I installed moguri mod 9.0, 30 fps FMV, and DualSense buttons. I checked 60 fps gameplay.

While watching the FMV, I noticed some stuttering. But when I played Vivi, it was a stuttering fest, especially when I made him run.

I tested the solution with the CPU affinity to one thread (with Process Lasso), but it made the CPU reach 100% on that thread, and the stuttering was still very much present. So I tested with 30 fps gameplay. On the same thread it could reach 60%, and it was never below 40%, which is not logical for a PS1 game. Furthermore, I still had some stuttering, but it was more bearable.

What can I do to solve this problem?

Thanks!

**Comment by SamsamTS** [comment]
That's not normal. I have a i5-9400F and a RTX 2060 and the game runs fine at 120fps. I have been told you are not the only one experiencing that issue but it's a mystery why it happens.

Looking at the i5-11500, it seems to have a integrated GPU (such a scam, with a dedicated GPU that's $100 that would have been better spent elsewhere). I would try disabling it, maybe Windows is using that GPU to run the game rather than the RTX 3090. [Here is a guide](https://www.howtogeek.com/disable-integrated-graphics-on-windows/) on how to do so.

**Comment by Neodoris** [comment]
Thank you for your help. I checked, and my iGPU was already disabled.
Here are my findings: at 60 fps, my RTX 3090 is used at 17% at 1080p, and the CPU is at 100% on one thread. If it's on two threads, they are used from 40 to 60% each, and the stuttering is still present.
I also tried tinkering with Triple Buffering, Vsync and Max Frame from nvidia control panel, and nothing worked, still experiencing a big stutter each second.
I'm confused.

What are your usage percentages with your components?

**Comment by SamsamTS** [comment]
Running around as Vivi in Alexendria at 120fps with Moguri enabled, I get around 25% CPU and 15% GPU usage.
![image](https://github.com/user-attachments/assets/37e2a4e7-0df3-4bd8-af5c-83e790150303)

I have really no idea what is causing this issue, sorry.

**Comment by SamsamTS** [comment]
Maybe you could help diagnose the issue with [Process Explorer](https://learn.microsoft.com/en-us/sysinternals/downloads/process-explorer). I would be interested to see which threads are hogging all the CPU.
![image](https://github.com/user-attachments/assets/8fb2df5f-1bdd-48b9-bf25-7a2100e2cc56)

**Comment by WarpedEdge** [comment]
Just to confirm. How does the game run vanilla?

**Comment by SamsamTS** [comment]
> Just to confirm. How does the game run vanilla?

8% CPU and 3% GPU. But the game is running at 30fps only, not in widescreen, without Moguri HD backgrounds, and no shaders.

**Comment by Neodoris** [comment]
I used Process Explorer and got that:
![2024-12-31 14_22_38-Greenshot](https://github.com/user-attachments/assets/26dc0ac9-5257-44a2-84b4-ad2635841aa1)

From this screenshot it seems there is no problem, so I decided to make some tests. Multithreading did not work initially, but I did not try Triple Buffering and Vsync with it.

So, I reverted back to multihtreading instead of leaving everything on one thread. I used nvidia control panel to add Triple buffering and Vsync, and finally, it works! Stuttering fest is gone.

There is sometimes a single stutter on big screens like the one with the place selling tickets, and the one which shows the tavern.

I do not know if Memoria will add an option for Triple buffering in the future, but as of now, my problem is gone.

Thank you for your help, I wish you an happy new year!

**Comment by SamsamTS** [comment]
>From this screenshot it seems there is no problem

Yeah, nothing useful here. Bummer, I would have like to know where the CPU hangs.

> So, I reverted back to multihtreading instead of leaving everything on one thread. I used nvidia control panel to add Triple buffering and Vsync, and finally, it works! Stuttering fest is gone.

That's some useful information for people that might experience the same issue. So triple buffering and/or Vsync seems to be the solution. Thanks for reporting back. You'd be amazed how many people you'll never hear from again once their problem is gone so we definitely appreciate it.

**Comment by WarpedEdge** [comment]
Thanks Neo, Samsam I meant vanilla for him to see when the problem
occurred, step by step to see where the stuttering happens, glad it fixed
for him.

**Comment by antonio-higino** [comment]
Firstly, sorry for reviving this closed issue, it's just that OP's last comment described the problem I'm having, so I felt it okay to comment. If I should instead open a new issue, no problem. 

> There is sometimes a single stutter on big screens like the one with the place selling tickets, and the one which shows the tavern.

That's the problem, here's what I tested and didn't work:

- Fullscreen, borderless, windowed.
- All FPS options.
- Disabling stable camera, widescreen, Moguri mod, absolute direction on analog controls.
- Set process affinity to 1 CPU thread (caused periodic stutter, thread was probably maxed out) and 2 threads/1 core (same behavior as normal CPU affinity).
- Setting GPU to high performance on NVIDIA control panel.
- Setting monitor to 60Hz instead of 120Hz.

Setup: 4070 Super, 7800X3D.
